### PR TITLE
Speaker attribute extraction

### PR DESCRIPTION
**Pull request by Aya-AlJafari**
Generate attribute-specific embeddings to condition TTS on. [Accents,gender,age,emotions,etc]

### CODE REVIEW

**Comment by p0p4k** [comment]
Hello, this is interesting. I am just trying to get deeper into the 🐸 TTS code base these days.

1. First, we convert '.wav' files into their respective embeddings, then this model converts them back to speaker labels? What is the reason for that?

2. Second. is the any particular reason for using Tensorflow, while rest of the codebase uses pytorch?

I have been learning pytorch recently and I tried to implement the same model in pytorch (didnt add cuda yet) , if it helps 🤷 
If this implementation is correct, we can modify the entire classifier file to use pytorch. 
```
import torch 
import torch.nn as nn
import torch.nn.functional as F
import torch.optim as optim

class baseline_model(nn.Module):
    def __init__(self, class_num):
        super().__init__()
        self.fc1 = nn.Linear(512, 1024)
        self.fc2 = nn.Linear(1024, 512)
        self.fc3 = nn.Linear(512, 128)
        self.fc4 = nn.Linear(128, 64)
        self.fc5 = nn.Linear(64, class_num)
        self.dropout1 = nn.Dropout(0.2)
        self.dropout2 = nn.Dropout(0.5)

    def forward(self, x):
        output=self.dropout1(F.relu(self.fc1(x)))
        output=self.dropout2(F.relu(self.fc2(output)))
        output=F.relu(self.fc3(output))
        output=F.relu(self.fc4(output))
        output=F.softmax(self.fc5(output),dim=0)
        return output


loss_fn = nn.CrossEntropyLoss()
optimizer = optim.Adam(model.parameters(), lr=0.0001)

class_num = 5
model = baseline_model(class_num)

epochs=10
for i in range(0,epochs):
    x=torch.randn(512) #random 512 dim embedding
    model.zero_grad()
    output = model(x)
    target =  torch.randint(class_num ,(1,)) #random target
    print(f'output : {output}, target : {target}')
    loss = loss_fn(output.unsqueeze(0),target=target)
    loss.backward()
    optimizer.step()
```

**Comment by erogol** [comment]
Can we do the same thing with Torch? Adding TF as a dependency is a big change.

**Comment by Aya-AlJafari** [comment]
yes, @erogol I'll push a Torch classifier.

**Comment by erogol** [comment]
We'd also need unit tests before we merge it. You can check the existing `tests` to have some idea of how we handle testing. Feel free to ping me too if you have more questions.

**Comment by erogol** [comment]
We could also use our trainer to have direct interfacing with the experiment tracking support. It should be pretty easy to update the training code. 

It'd also help unit testing like https://github.com/coqui-ai/TTS/blob/dev/tests/tts_tests/test_vits.py

You can check https://github.com/coqui-ai/Trainer/blob/main/tests/utils/mnist.py and https://github.com/coqui-ai/Trainer/blob/main/tests/utils/train_mnist.py as a tutorial.

**Review by erogol — CHANGES_REQUESTED** [review]

File: spkr-attr/generate_pseudo_spkrs/pool_selection.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    if len(average_dvector) == 0 :
+        print(f"Not enough speakers for {val}, Skipping!")
+        return 
+    outputs = synthesis(model=tts_model,
+                    text="Once upon a time, the King’s youngest son became filled with the desire to go abroad and see the world.",
+                    CONFIG=tts_config, use_cuda=False,d_vector=average_dvector,language_id=0)
+    waveform = outputs["wav"]
+    scipy.io.wavfile.write(f"{out_path}/{label}-{val}-{num_spkrs}.wav", 44100, waveform)
+
+def main():
```
Comment (erogol) [inline spkr-attr/generate_pseudo_spkrs/pool_selection.py:71]: I think it is better to make this script a class or usable through import in some other way.

In its current form, it cannot be used if it is installed by ```pip install TTS```

**Review by erogol — CHANGES_REQUESTED** [review]
